CMake error messages when ACE absent are less helpful than they could be

When CMake is run on a fresh checkout of YARP, with ACE, absent, it gives this error:

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
ACE_ACE_LIBRARY
    linked by target "YARP_OS" in directory /.../yarp/src/libYARP_OS
    linked by target "YARP_sig" in directory /.../yarp/src/libYARP_sig
    ...
 ACE_INCLUDE_DIR (ADVANCED)
   used as include directory in directory /.../yarp/src/libYARP_OS
   used as include directory in directory /.../yarp/src/libYARP_sig
   ... 
```

It would be clearer to state that a library called "ACE" was not found, and perhaps offer solutions -- either to install it, or set SKIP_ACE.  We could also consider making ACE optional on Linux/OSX, but we'd want to communicate well about that since user loses some carriers not yet implemented without ACE.

--- a/conf/YarpFindDependencies.cmake
+++ b/conf/YarpFindDependencies.cmake
@@ -120,6 +120,18 @@ macro(check_optional_dependency optionname package)
 endmacro(check_optional_dependency)
 
 
+# Check if a dependency required to disable an option is installed.
+macro(check_skip_dependency optionname package)
+    string(TOUPPER ${package} PKG)
+
+    if(NOT ${optionname})
+        if(NOT YARP_HAS_${PKG})
+            message(FATAL_ERROR "Optional package ${package} not found. Please install it or enable the option \"${optionname}\" to build yarp.")
+        endif()
+    endif()
+endmacro()
+
+
 # Print status for a dependency
 macro(print_dependency package)
 
@@ -270,6 +282,7 @@ endif(CREATE_GYARPBUILDER)
 
 # CHECK DEPENDENCIES:
 
+check_skip_dependency(SKIP_ACE ACE)
 check_required_dependency(SQLite)
 check_optional_dependency(CREATE_LIB_MATH GSL)
 check_optional_dependency(YARP_USE_ATLAS Atlas)
@@ -286,6 +299,7 @@ check_optional_dependency(CREATE_GYARPBUILDER GooCanvasMM)
 
 message(STATUS "I have found the following libraries:")
 
+print_dependency(ACE)
 print_dependency(SQLite)
 print_dependency(GSL)
 print_dependency(Atlas)

